### ISSUE TITLE
espanso disabling it self and not turning on again ? 

### ISSUE THREAD

**Issue by maxandersen** [federico-terzi/espanso#238]
after #216 i installed latest and now seeing the warning notficiation when browsers and others enables secure input which is nice.

What I have also noticed though is that espanso seem to eventually turn it self off and I have to go enable it again.

Is there by any chance a bug where espanso due to secure input disables itself and then does not realize it can start again or ?

**Comment by federico-terzi** [comment]
Hey @maxandersen ,

Thanks for the feedback, that seems unlikely but I'll take a better look at it.
Are you sure you are not accidentally pressing the `toggle_key`? (by default it's option on macOS)

Cheers :)

**Comment by federico-terzi** [comment]
Also, if you experience problems, check out the `espanso log`.
After an application releases SecureInput, a "Secure Input disabled" log should be recorded.
If you see an app that requested the SecureInput without releasing it, that line should be missing.

If you experience again this problem, could you please post the log here?

**Comment by vexillia** [comment]
I'm seeing this with xubuntu 20.04 and v0.6.  I have to manually restart espanso at least once a day.  Log file shows nothing but restart information.

**Comment by federico-terzi** [comment]
> I'm seeing this with xubuntu 20.04 and v0.6. I have to manually restart espanso at least once a day. Log file shows nothing but restart information.

@vexillia I'm sorry you experienced this issue. Could you please post your log here?

**Comment by vexillia** [comment]
This is my current log after I restarted espanso this morning:

08:36:53 [ INFO] espanso version 0.6.0
08:36:53 [ INFO] using config path: /home/vexillia/.config/espanso
08:36:53 [ INFO] using package path: /home/vexillia/.local/share/espanso/packages
08:36:53 [ INFO] spawning worker process...
08:36:53 [ INFO] Binded to IPC unix socket: /home/vexillia/.local/share/espanso/espanso.sock
08:36:53 [ INFO] initializing worker process...
08:36:53 [ INFO] Binded to IPC unix socket: /home/vexillia/.local/share/espanso/worker.sock
08:36:53 [ INFO] watching for changes in path: /home/vexillia/.config/espanso
08:36:54 [ INFO] worker is running!

Still running OK since this morning.

**Comment by federico-terzi** [comment]
All right @vexillia, this log looks perfect. If you can, post again the log the next time espanso is not working **before restarting it**.

Cheers :)

**Comment by vexillia** [comment]
Log after failure to autostart after reboot:

13:28:41 [ INFO] espanso version 0.6.0
13:28:41 [ INFO] using config path: /home/vexillia/.config/espanso
13:28:41 [ INFO] using package path: /home/vexillia/.local/share/espanso/packages
13:28:41 [ INFO] spawning worker process...
13:28:41 [ INFO] Binded to IPC unix socket: /home/vexillia/.local/share/espanso/espanso.sock
13:28:41 [ INFO] initializing worker process...
13:28:41 [ERROR] Error, can't connect to X11 context
13:28:41 [ INFO] watching for changes in path: /home/vexillia/.config/espanso

Hope this helps.

**Comment by vexillia** [comment]
Could this be a timing issue?  If, after espanso doesn't work, I run"espanso start" I get "espanso already running" but it doesn't work.  However "espanso restart" works fine and expansions work properly.

Just a  thought.

**Comment by federico-terzi** [comment]
Hey @vexillia,

Thank you for the log, this seems to be a new bug introduced in the last version! I've opened a new issue for that, so make sure to follow #284 for further updates.

Cheers :)
